### PR TITLE
Refactor ActionInstructionReceiverImpl to improve clarity

### PR DESCRIPTION
**Pull request by tomphp**
# Motivation and Context
This commit intends to make it easier to understand the code.
This commit makes not changes to the functionality of the system.

# What has changed
This PR only contains refactorings, no functionality has changed.
- [Extract Method](https://refactoring.com/catalog/extractMethod.html) to improve readability
- [Replace Nested Conditional with Guard Clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)
- [Replace Method with Object](https://refactoring.com/catalog/replaceMethodWithMethodObject.html)

# How to test?
Run the acceptance tests.

# Links
Done as part of https://trello.com/c/lZYrIXw7/118-create-an-auto-verifying-notify-endpoint-service

### CODE REVIEW

**Comment by PurpleBooth** [comment]
Is Coday misconfigured? I was expecting to see Cyclomatic complexity, Npath, coupling metrics, method size, class size alerts, but they seem to be missing?

File: src/main/java/uk/gov/ons/ctp/response/notify/message/impl/ActionInstructionReceiverImpl.java [hunk 1]
```diff
@@ -46,112 +46,146 @@
...
+
+  private String getNotifySMSNotSent() {
+    return truncateToSituationMaxLength(NOTIFY_SMS_NOT_SENT);
+  }
+
+  private String getNotifyGateway() {
+    return truncateToSituationMaxLength(NOTIFY_GW);
+  }
+
+  private String truncateToSituationMaxLength(final String string) {
```
[the remark this answers is not in the document]
Comment (tomphp) [reply src/main/java/uk/gov/ons/ctp/response/notify/message/impl/ActionInstructionReceiverImpl.java]: Great spot! I guess my concern would be protection against any of the constants changing, but that's never going to happen right ;)

The correct approach should be that this is handled in the `ActionFeedback` class, we did discuss that but didn't want to add too much complexity to this PR - I'd prefer to do it as a separate one.

File: src/main/java/uk/gov/ons/ctp/response/notify/message/impl/ActionInstructionReceiverImpl.java [hunk 2]
```diff
@@ -46,112 +46,146 @@
...
+  private class ResponsePublisher {
+    private final boolean responseRequired;
+
+    ResponsePublisher(final boolean responseRequired) {
+      this.responseRequired = responseRequired;
     }
 
-    return result;
+    private void sendFeedback(final ActionFeedback actionFeedback) {
+      if (responseRequired) {
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/notify/message/impl/ActionInstructionReceiverImpl.java]: this.responseRequired?

**Comment by tomphp** [comment]
[Create an auto-verifying notify endpoint service](https://trello.com/c/lZYrIXw7/118-create-an-auto-verifying-notify-endpoint-service)

**Pull request merged**